### PR TITLE
2020 Strategic Implementation Plan for Xdev

### PR DESCRIPTION
**Pull request by kmpaul**
I have attempted to write up the results of the Xdev Start/Stop/Continue exercise conducted on July 1, 2020, into a Strategic Plan document for 2020.  When possible, I have attempted to sketch out, based on input from the group, a plan for each *Activity* touched in the Start/Stop/Continue exercise.

@NCAR/xdev:  I would really appreciate your reviews and input!

### CODE REVIEW

File: strategy/plan2020.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+     - Determine how to obtain the cheapest deployment possible for various purposes:
+       education/training, science/research, development/testing
+     - A specialized hub for online tutorials/self-paced learning
+     - A small-team research deployment (e.g., for Matt Long’s group, or for us
+       to use to test notebooks and notebook scaling solutions)
+     - Target a new service to implement that would need a testing/development hub
+       (e.g., a notebook sharing/publication repository)
+   - Determine best use cases (1-2) and deploy hubs for those use cases
+   - Blog post
+5. Xdev Communication Strategy [4+1 points]
```
Comment (jukent) [inline strategy/plan2020.md:140]: Is "4+1 points" different from 5? Not sure how to read that.

Comment (kmpaul) [reply strategy/plan2020.md:140]: Yeah.  I try to explain that in the next section, but that's bad practice on my part.  I should explain it in the intro text at the top of the section.

Comment (kmpaul) [reply strategy/plan2020.md:140]: ...Will fix.

Comment (kmpaul) [reply strategy/plan2020.md:151]: @jukent Take a look at it, now.

File: strategy/plan2020.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+- **Start:** What activities are we *not* currently engaged in
+  that we absolutely *should* be engaged?
+- **Stop:** What activities are we currently engaged in that we
+  should *stop* doing immediately.
+- **Continue:** What activities are we currently engaged in that
+  we should endeavor to continue.
+
+After 30 minutes of brainstorming ideas in each category above,
+each participating member of the Xdev Team was given 5 points to
+distribute across activites listed in the *Start* and *Stop*
```
[the remark this answers is not in the document]
Comment (kmpaul) [reply strategy/plan2020.md:18]: Ooh!  Good catch!

**Comment by kmpaul** [comment]
Thanks, @sherimickelson!

File: strategy/plan2020.md (hunk 1 above)
Comment (jukent) [reply strategy/plan2020.md:151]: Makes sense!

File: strategy/plan2020.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+categories, essentially given everyone on the Team the ability
+to vote for what *changes* should be taken in the future of our
+work.  The results of this exercise are detailed below.
+
+## Strategic Activities
+
+In our own terminology, the Xdev Team's purpose/mission can be
+categorized into two different *Campaigns*: *Workflows* and
+*Outreach, Education & Community Building*.  As a team, the work
+we do specifically engages in short-term *Projects* targeting
```
Comment (halehawk) [inline strategy/plan2020.md:28]: Should engages be engage?

Comment (kmpaul) [reply strategy/plan2020.md:28]: I don't think so.  I think the tense is corrent.  If you cut out the "filler" text, it should read:

> ...the work we do engages in short-term *Projects*...

I think that's correct.

File: strategy/plan2020.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+   - Add notebooks to the Pangeo Gallery (gallery.pangeo.io)
+   - Identify examples that make good blog posts, too!
+4. Python Tutorials
+   - Take tutorials virtual until COVID is not a risk
+   - Grow online, self-paced content
+
+### New Activities
+
+These are Activities in which the Xdev Team plan to engage in the
+next year (July 2020 - June 2021).  Each individual activities shows
```
Comment (halehawk) [inline strategy/plan2020.md]: activities -> activity

Comment (kmpaul) [reply strategy/plan2020.md]: Good catch!  Will fix.

Comment (kmpaul) [reply strategy/plan2020.md]: Actually, you are looking at an old revision.  You might need to refresh.  That error has been fixed already.

**Comment by halehawk** [comment]
It's great, I cannot find any typo.

**Comment by kmpaul** [comment]
Thanks, @halehawk!

File: strategy/plan2020.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+     notebooks hosted on GitHub repositories along with the notebook environments
+
+### Terminating Activities
+
+The Activities listed in the *Stop* category of this exercise, for the most
+part, corresponded to Activities listed in the *Start* category, above.  That is,
+each Activity identified as something we should *Stop* had a "mirror" Activity
+that we identified as something we should *Start*.  For example, "Start better
+communication practices" mirrored "Stop bad communication practices".  Hence,
+there is no separate list of Activities to terminate.
```
[the remark this answers is not in the document]
Comment (kmpaul) [reply strategy/plan2020.md:219]: Do you think I should just delete this section, then?

File: strategy/plan2020.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+     - Determine how to obtain the cheapest deployment possible for various purposes:
+       education/training, science/research, development/testing
+     - A specialized hub for online tutorials/self-paced learning
+     - A small-team research deployment (e.g., for Matt Long’s group, or for us
+       to use to test notebooks and notebook scaling solutions)
+     - Target a new service to implement that would need a testing/development hub
+       (e.g., a notebook sharing/publication repository)
+   - Determine best use cases (1-2) and deploy hubs for those use cases
+   - Blog post
+5. Xdev Communication Strategy [4+1 points]
```
[the remark this answers is not in the document]
Comment (kmpaul) [reply strategy/plan2020.md:151]: Thanks!  I tried to gather all the thoughts I heard and put them together.  I hope it is all clear to everyone.

File: strategy/plan2020.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mnlevy1981) [reply strategy/plan2020.md:219]: I'm not sure what the best remedy is. Maybe keep the section, but have the text be something like

> As mentioned above, each Activity identified as something we should *Stop* had a "mirror" Activity that we identified as something we should *Start*. Hence, there is no separate list of Activities to terminate.

perhaps with a link to the New Activities section?

Another approach would be to keep the section and duplicate the Activities that were mirrored above, but I don't think that would be very useful. Is there a third option I'm missing?

[the remark this answers is not in the document]
Comment (kmpaul) [reply strategy/plan2020.md:219]: I like this solution (your initial proposal) with the link.  I'll implement that.

**Comment by kmpaul** [comment]
Thanks, @mnlevy1981!

**Comment by kmpaul** [comment]
Thanks, @jukent!

**Comment by kmpaul** [comment]
Thanks, @halehawk!

**Comment by kmpaul** [comment]
@bonnland I think both of those ideas are represented in this plan.  I like the badge idea.  That would be a very simple heroku app, for example.  And the data upload examples definitely fits with the current ideas on EOCB.

**Comment by andersy005** [comment]
@bonnland, I am curious.... Does Dash (data.ucar.edu) provide any end-user guide or developer guide documentation? I'd be happy to help out with the second point, but I don't really understand how dash works.

**Comment by bonnland** [comment]
> Does Dash (data.ucar.edu) provide any end-user guide or developer guide documentation? I'd be happy to help out with the second point, but I don't really understand how dash works.

The core application (CKAN) documentation is here:   http://docs.ckan.org/en/latest/.   We are using version 2.8 of CKAN at the moment.

It is a plugin-based architecture, currently in transition from pylons to flask, and python 2.7 to 3.   The HTML templates are based on jinja2.   Currently, our application does not reference any external sites, but there are mechanisms for doing so safely. 

Our custom plugin for DASH Search is here:  https://github.com/NCAR/ckanext-dsetsearch

I would welcome the chance to work on this together.

**Comment by kmpaul** [comment]
@NCAR/xdev:  Unless there are any objections, I'll merge this at the end of the day.

**Comment by kmpaul** [comment]
Thanks, everyone!

**Pull request merged**